src/Nethermind/Nethermind.State.Test/StorageProviderTests.cs

```diff
@@ -185,20 +185,17 @@ public void Commit_trees_clear_caches_get_previous_root()
         WorldState storageProvider = BuildStorageProvider(ctx);
         storageProvider.Set(new StorageCell(ctx.Address1, 1), _values[1]);
         storageProvider.Commit(Frontier.Instance);
-
-        ctx.StateProvider.Commit(Frontier.Instance);
-        ctx.StateProvider.CommitTree(0);
+        storageProvider.Commit(Frontier.Instance);
```

Comment:
Although this is a double commit, so shouldn't do anything; it did help find a bug, so keeping

Reply:
The difference is in the order. CommitTree applies to the tree, so if you commit first, you push the changes in the tree to db and later apply it. Usually it's the other way around.

Reply:
🤔 Commit doesn't apply to db; moves the "checkpoint" so how far a restore can go, then CommitTree applies to db